src/core_codemods/harden_pyyaml.py

```diff
@@ -50,18 +53,80 @@ def rule(cls):
+                                - pattern: yaml.BaseLoader
+                                - pattern: yaml.FullLoader
+                                - pattern: yaml.UnsafeLoader
 
         """
 
-    def on_result_found(self, original_node, updated_node):
-        maybe_name = self.get_aliased_prefix_name(original_node, self._module_name)
-        maybe_name = maybe_name or self._module_name
-        if maybe_name == self._module_name:
-            self.add_needed_import(self._module_name)
-        new_args = [
-            *updated_node.args[:1],
-            updated_node.args[1].with_changes(
-                value=self.parse_expression(f"{maybe_name}.SafeLoader")
-            ),
+    def on_result_found(
+        self,
+        original_node: Union[cst.Call, cst.ClassDef],
+        updated_node: Union[cst.Call, cst.ClassDef],
+    ):
+        # TODO: provide different change description for each case.
+        match original_node:
+            case cst.Call():
+                maybe_name = self.get_aliased_prefix_name(
+                    original_node, self._module_name
+                )
+                maybe_name = maybe_name or self._module_name
+                if maybe_name == self._module_name:
+                    self.add_needed_import(self._module_name)
+                new_args = [
+                    *updated_node.args[:1],
+                    updated_node.args[1].with_changes(
+                        value=self.parse_expression(f"{maybe_name}.SafeLoader")
+                    ),
+                ]
+                return self.update_arg_target(updated_node, new_args)
+            case cst.ClassDef():
+                return updated_node.with_changes(
+                    bases=self._update_bases(original_node)
+                )
+        return updated_node
+
+    def _update_bases(self, original_node: cst.ClassDef) -> list[cst.Arg]:
+        new = []
+        unsafe_name_matchers = (
+            matchers.Name(value="UnsafeLoader")
+            | matchers.Name(value="Loader")
+            | matchers.Name(value="BaseLoader")
+            | matchers.Name(value="FullLoader")
+        )
+        base_names = [
+            f"yaml.{klas}"
+            for klas in ("UnsafeLoader", "Loader", "BaseLoader", "FullLoader")
         ]
-        return self.update_arg_target(updated_node, new_args)
+        for base_arg in original_node.bases:
+            base_name = self.find_base_name(base_arg.value)
+            if base_name not in base_names:
+                new.append(base_arg)
+                continue
+
+            if matchers.matches(base_arg.value, unsafe_name_matchers):
+                self.add_needed_import(self._module_name, "SafeLoader")
+                self.remove_unused_import(base_arg.value)
+                base_arg = base_arg.with_changes(
+                    value=base_arg.value.with_changes(value="SafeLoader")
+                )
+            if matchers.matches(
```

Comment:
It took me awhile to understand why this code works, even though the tests are passing. I think this can all be simplified a bit more for clarity:
```python
for base_arg in original_node.bases:
    # Ensures that we only consider valid names
    base_name = self.find_base_name(base_arg.value)
    if base_name not in base_names:
        new.append(base_arg)
        continue

    match base_arg.value:
        case cst.Name():
            self.add_needed_import(self._module_name, "SafeLoader")
            self.remove_unused_import(base_arg.value)
            base_arg = base_arg.with_changes(
                 value=base_arg.value.with_changes(value="SafeLoader")
            )
        case cst.Attribute():
            base_arg = base_arg.with_changes(
                value=base_arg.value.with_changes(attr=cst.Name("SafeLoader"))
            )

    new.append(base_arg)
```